tests/unittests/tests-ipv6_addr/tests-ipv6_addr.c

```diff
@@ -0,0 +1,619 @@
+/*
+ * Copyright (C) 2015 Martine Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @{
+ *
+ * @file    tests-ipv6_addr.c
+ */
+#include <errno.h>
+#include <stdint.h>
+
+#include "embUnit/embUnit.h"
+
+#include "byteorder.h"
+#include "net/ng_ipv6/addr.h"
+
+#include "tests-ipv6_addr.h"
+
+static void test_ipv6_addr_equal_not_equal(void)
+{
+    ng_ipv6_addr_t a = { {
+            0x00, 0x01, 0x02, 0x03, 0x04, 0x05, 0x06, 0x07,
+            0x08, 0x09, 0x0a, 0x0b, 0x0c, 0x0d, 0x0e, 0x0f
+        }
+    };
+    ng_ipv6_addr_t b = { {
+            0x10, 0x11, 0x12, 0x13, 0x14, 0x15, 0x16, 0x17,
+            0x18, 0x19, 0x1a, 0x1b, 0x1c, 0x1d, 0x1e, 0x1f
+        }
+    };
+    TEST_ASSERT_EQUAL_INT(false, ng_ipv6_addr_equal(&a, &b));
```

Comment:
Try with different IP addresses. E.g. with addresses that only differs at the first, the last or an arbitrary bit or differs only for two bits.